packages/merge-styles/src/Stylesheet.ts

```diff
@@ -154,9 +192,28 @@ export class Stylesheet {
    * Configures a reset callback.
    *
    * @param callback - A callback which will be called when the Stylesheet is reset.
+   * @returns function which when called un-registers provided callback.
    */
-  public onReset(callback: () => void): void {
+  public onReset(callback: Function): Function {
```

Comment:
In order to prevent memory leaks by holding these callbacks and their entire closure indefinitely (which would happen because we are a singleton) we now return an `unregister` function.